numpy/ma/core.py

```diff
@@ -4287,22 +4298,44 @@ def count(self, axis=None):
         array([3, 0])
 
         """
+        kwargs = {}
+        if keepdims is not np._NoValue:
+            kwargs['keepdims'] = keepdims
+
         m = self._mask
-        s = self.shape
-        if m is nomask:
+        # optimization if we exactly have an ndarray
+        if m is nomask and type(self._data) is np.ndarray:
+            # see _count_reduce_items in _methods.py
+            axes = axis
```

Comment:
Somewhat of a detail, but my preference would be something more readable, perhaps as follows:
```
if self.shape is ():
    return 1
elif axis is None:
    return self.size

axes = axis if isinstance(axis, tuple) else (axis,)
items = 1
for ax in axes:
    items * self.shape[ax]

if kwargs.get('keepdims', False):
...
```